doc/source/contributing_docstring.rst

```diff
@@ -0,0 +1,634 @@
+            """Generate a plot.
+
+            Render the data in the Series as a matplotlib plot of the
+            specified kind.
+
+            Parameters
+            ----------
+            kind : str
+                Kind of matplotlib plot.
+            **kwargs
+                These parameters will be passed to the matplotlib plotting
+                function.
+            """
+            pass
+
+**Bad:**
+
+.. code-block:: python
+
+    class Series:
+        def plot(self, kind, **kwargs):
+            """Generate a plot.
+
+            Render the data in the Series as a matplotlib plot of the
+            specified kind.
+
+            Note the blank line between the parameters title and the first
+            parameter. Also, not that after the name of the parameter `kind`
+            and before the colo, a space is missing.
+
+            Also, note that the parameter descriptions do not start with a
+            capital letter, and do not finish with a dot.
+
+            Finally, the `**kwargs` parameter is missing.
+
+            Parameters
+            ----------
+
+            kind: str
+                kind of matplotlib plot
+            """
+            pass
+
+Parameter types
+^^^^^^^^^^^^^^^
+
+When specifying the parameter types, Python built-in data types can be used
+directly:
+
+- int
+- float
+- str
+
+For complex types, define the subtypes:
+
+- list of [int]
+- dict of {str : int}
+- tuple of (str, int, int)
+- set of {str}
+
+In case there are just a set of values allowed, list them in curly brackets
+and separated by commas (followed by a space):
+
+- {0, 10, 25}
+- {'simple', 'advanced'}
+
+If the type is defined in a Python module, the module must be specified:
+
+- datetime.date
+- datetime.datetime
+- decimal.Decimal
+
+If the type is in a package, the module must be also specified:
+
+- numpy.ndarray
+- scipy.sparse.coo_matrix
+
+If the type is a pandas type, also specify pandas:
+
+- pandas.Series
+- pandas.DataFrame
+
+If more than one type is accepted, separate them by commas, except the
+last two types, that need to be separated by the word 'or':
+
+- int or float
+- float, decimal.Decimal or None
+- str or list of str
+
+If None is one of the accepted values, it always needs to be the last in
+the list.
+
+Section 4: Returns or Yields
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If the method returns a value, it will be documented in this section. Also
+if the method yields its output.
+
+The title of the section will be defined in the same way as the "Parameters".
+With the names "Returns" or "Yields" followed by a line with as many hyphens
+as the letters in the preceding word.
+
+The documentation of the return is also similar to the parameters. But in this
+case, no name will be provided, unless the method returns or yields more than
+one value (a tuple of values).
+
+The types for "Returns" and "Yields" are the same as the ones for the
+"Parameters". Also, the description must finish with a dot.
```

Comment:
Do we want to require a description?

Because as a dummy example for pivot:

```
Pivot dataframe.

Longer explanation ..

parameters ...

Returns
-------
DataFrame
    Pivoted dataframe.
```

it can be quite redundant in some cases I think. 
I would say this is only required when some clarification is needed or when there are multiple return values

Reply:
I'd start requiring them, and if we see examples where it doesn't add any value, I'd make it optional, and clarify when can be omitted based on the examples.